### ISSUE TITLE
Is this plugin ready to use?

### ISSUE THREAD

**Issue by ptgamr** [Kinto/kinto-webpush#3]
If not, when can we use it? As I would like to integrate my own websocket rather than using pusher.

**Comment by Natim** [comment]
Hello @ptgamr this is almost ready to use, the only missing piece is the listener but we can finish it sooner than later next week.

**Comment by ptgamr** [comment]
Thank @Natim , it's a great news ;) I'm writing an Ember Data adapter for Kinto & intended to write a blog post about this combination. Hopefully this plugin will be ready by then ;-)

**Comment by lgleim** [comment]
Any updates on this?

**Comment by Natim** [comment]
Yes we did a bunch of work that can be find here: https://github.com/Kinto/kinto-webpush/tree/add-listener

The problem being that when you do a pattern matching on the parent_id with the storage, you have no way of knowing the parent_id in the response. That's something that needs to be fixed in the Kinto storage backend.

**Comment by leplatrem** [comment]
Why do you need the `parent_id` ? (with the current code, it stores the `userid` of the subscription)

Is there an opened issue on the server to track?

**Comment by Natim** [comment]
We need the `user_id` to check the permission if I recall correctly.
We want to check that the user has got the permission to get notified.
I don't think we have an issue on the kinto yet because I was trying to figure out another way to store the data before. But now that I think about it it looks like the way to go.

I mean if you don't know the userid and you use a wildcard you should have a way to get it back.
